### ISSUE TITLE
Chinese _matchesInfo length

### ISSUE THREAD

**Issue by tyuan511** [meilisearch/meilisearch#2149]
**Describe the bug**

_matchesInfo length seems like a byte length, so one Chinese character will return 3

same issue with cropLength

**To Reproduce**
```
import { MeiliSearch } from 'meilisearch';

const client = new MeiliSearch({
  host: 'http://127.0.0.1:7700',
  apiKey: 'hellomat2021',
});

const index = client.index('test');
const documents = [{ id: 1, title: '一二三四' }];
await index.addDocuments(documents);

const res = await index.search('一', { matches: true });
console.log(res);
```

_matchesInfo will return 
```
"title": [
  {
    "start": 0,
    "length": 3
  }
]
```

**Expected behavior**
```
"title": [
  {
    "start": 0,
    "length": 1
  }
]
```

**Comment by Kerollmops** [comment]
Hey @tyuan511,

Which version of Meilisearch do you use? Just to know the version of the tokenizer that returned those boundaries.
However, I'm pretty sure that the length is in bytes and not in character count, you can verify this on [this website](https://mothereff.in/byte-counter) for example i.e. `一` is 3 bytes long.

**Comment by tyuan511** [comment]
I use Meilisearch v0.25.2.
Yes,  it's  bytes length . I just want to highlight content by the _matchesInfo, but it's hard to slice a string by byte  for javascript in browser.

**Comment by gmourier** [comment]
Hi @tyuan511 👋

Why did you want to highlight from `_matchesInfo` and not `_formatted`? 

Thanks!

**Comment by tyuan511** [comment]
I used to _formatted with cropLength, but cropLength is also in  bytes (e.g. cropLength 50 will only return 17 Chinese character).  I think I can use _formatted without cropLength, and slice the result by html em tag. 

Thanks!

**Comment by Sampwood** [comment]
Yes, for article mixed with Chinese and English, it is difficult to control the length of the highlight.

**Comment by miiton** [comment]
I think `Token` object should have `char_start` and `char_end`  in meilisearch/tokenizer  for include these to `_matchesInfo`

https://github.com/meilisearch/tokenizer/blob/491071ab186d8d240dd105a4a29f04d4429a225d/src/token.rs#L28-L40

And Jieba also has unicode based `start` and `end` property, so this doesn't seem to be that hard to implementing.

https://github.com/messense/jieba-rs/blob/2a093182f254f5e96997f133bcda4f11186c0620/src/lib.rs#L186-L193

**Comment by curquiza** [comment]
Hello everyone

Looks like there are multiples issues in this issue 🙂
- Regarding `_matchesInfo` that returns bytes instead of chars -> this is expected, so not a bug. However this is something we are considering to change. Please check out [this discussion](https://github.com/meilisearch/product/discussions/393), and if you have the time, you can let your feedback regarding your usage and your need. This will definitely help the product team making a decision :)
- Regarding the bad cropping (cf @tyuan511 comment "cropLength is also in bytes (e.g. cropLength 50 will only return 17 Chinese character)") -> this is not something expected. Could you please @tyuan511 open another issue detailing the problem you observe with the cropping and how to reproduce it? We plan to stabilize the cropping part in the near future, so any feedback and bug reports are more than welcome 🙏 
- Regarding any bad highlighting -> we've just released the first release candidate of v0.26.0 that should fix the highlight with non-unicode char. You can check it using [v0.26.0rc0](https://github.com/meilisearch/meilisearch/releases/tag/v0.26.0rc0). If you have an issue with v0.26.0rc0, please open another issue

I close this issue since the original topic is _matchesInfo and for every point I listed, the discussion should continue in another place (in the discussion or in another fresh new issue) 🙂
